src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -154,6 +178,32 @@ public boolean hasWriteConnection() {
     }
 
     public void close() throws SQLException {
+        if (compatibleWithPreviousVersion) {
+            close_old();
+        } else {
+            close_new();
+        }
+    }
+
+    public void close_new() throws SQLException {
```

Comment:
Don't we want to have new test cases for these?

Reply:
Test cases are green for both old and new, and _old (we don't have repro) will be removed just after the release.